### DISCUSSION TITLE
Patching: Update value of a field with current metadata value (RQL)

### DISCUSSION THREAD

**Discussion by Fred-Richer** [ravendb/ravendb#16640] [Q&A] [answered]
Hi, I'm trying to add a new metadata field to a document type. It needs to be initialized with the current value of @last-modified and will be updated client side each time the document is loaded and modified. The client side part is working I'm just trying to find a way to patch the documents ideally with RQL from the studio.

Here's what I've tried so far:
`from "Quotations" q
update {
    q["@metadata"]["Modified-Date"] = q.@metadata.@last-modified    
}
`

AND

`from "Quotations" q
update {
    q["@metadata"]["Modified-Date"] = q["@metadata"]["@last-modified"]    
}`

It's the right part of the assignation that is not working since changing it with anything else does the assignation correctly.

Is there a way to access the current document's metadata (for reading) within the "update" statement or from outside as long as the value is reachable from within the update statement.

Thanks!

**Comment by ayende** [comment, marked as answer]
Try:

```
from "Quotations" q update 
{
           q["@metadata"]["Modified-Date"] = lastModified(q)
}

```

**Reply by Fred-Richer** [reply]
Thanks!

**Reply by Fred-Richer** [reply]
@ayende Is there a way to format the DateTime to: "2023-06-12T19:57:32.7060000Z" instead of this : "1686588006279.3975"?

**Reply by ayende** [reply]
`toISOString()`

**Reply by Fred-Richer** [reply]
Thanks @ayende but I actually just needed to instantiate a new date from the lastModified(q) result:

`from "Quotations" as q
where true and not exists(q.ModifiedDate)
update {
    q.ModifiedDate = new Date(lastModified(q))      
}`
